lib/openstudio-standards/standards/necb/NECB2011/building_envelope.rb

```diff
@@ -726,44 +728,116 @@ def apply_max_fdwr_nrcan(model:, fdwr_lim:)
-    # determined by the SRR.  The name of the skylight will be the surface name with the subsurface type attached
-    # ('skylight' in this case).  Note that this method will only work if the surface does not fold into itself (like an
-    # L or a V).
-    exp_surf_info['exp_nonplenum_roofs'].sort.each do |roof|
-      # sub_surface_create_centered_subsurface_from_scaled_surface(roof, srr_lim, model)
-      sub_surface_create_scaled_subsurfaces_from_surface(surface: roof, area_fraction: srr_lim, construction: skylight_construct_set)
+      # Go through all of exposed roofs adjacent to heated, non-plenum spaces, remove any existing subsurfaces, and add
+      # a skylight in the centroid of the surface, with the same shape of the surface, only scaled to be the area
+      # determined by the SRR.  The name of the skylight will be the surface name with the subsurface type attached
+      # ('skylight' in this case).  Note that this method will only work if the surface does not fold into itself (like an
+      # L or a V).
+      exp_surf_info['exp_nonplenum_roofs'].sort.each do |roof|
+        # sub_surface_create_centered_subsurface_from_scaled_surface(roof, srr_lim, model)
+        sub_surface_create_scaled_subsurfaces_from_surface(surface: roof, area_fraction: srr_lim, construction: skylight_construct_set)
+      end
+    else # OPTION B
+      spaces = model.getSpaces
+      types  = model.getSpaceTypes
+      roofs  = TBD.facets(spaces, "Outdoors", "RoofCeiling")
+
+      # A model's 'nominal' gross roof area (gra0) may be greater than its
+      # 'effective' gross roof area (graX). For instance, the "SmallOffice"
+      # prototype model has (unconditioned) attic roof overhangs that end up
+      # tallied as a gross roof area in OpenStudio and EnergyPlus. See:
+      #
+      #   github.com/rd2/osut/blob/117c7dceb59fd8aab771da8ba672c14c97d23bd0
+      #   /lib/osut/utils.rb#L6268
+      #
+      gra0 = roofs.sum(&:grossArea)    # nominal gross roof area
+      graX = TBD.grossRoofArea(spaces) # effective gross roof area
+
+      unless gra0.round > 0
+        msg = 'Invalid nominal gross roof area. No skylights will be added.'
+        OpenStudio.logFree(OpenStudio::Info, 'openstudio.model.Model', msg)
+        return false
+      end
+
+      unless graX.round > 0
+        msg = 'Invalid effective gross roof area. No skylights will be added.'
+        OpenStudio.logFree(OpenStudio::Info, 'openstudio.model.Model', msg)
+        return false
+      end
+
+      self.osut[:gra0] = gra0
+      self.osut[:graX] = graX
+
+      # Relying on the total area of attic roof surfaces (for SRR%) exagerrates
+      # the requested skylight area, often by 10% to 15%. This makes it unfair
+      # for NECBs, and more challenging when dealing with skylight wells. This
+      # issue only applies with attics - not plenums. Trim down SRR if required.
+      target = (srr_lim * graX / gra0) * graX
```

Comment:
_OSut_'s _addSkylights_ yields better results if smaller utility/technical rooms (e.g. narrow corridors, WCs) are first set aside. If feasible, _addSkylights_ prioritizes larger spaces that aren't daylit.

<img width="819" alt="hospital" src="https://github.com/user-attachments/assets/00d90688-44a0-426b-a928-10517f07c589">

_Hospital prototype_

For this reason, it is preferable to send as argument the actual SRR-based skylight area (`addSkyLights(spaces, {area: target})`), rather than the initial SRR (`addSkyLights(spaces, {srr: 0.05)`). In both example _addSkylights_ calls here, `spaces` is a subset of `model.getSpaces`.

_Gross roof area_ (GRA) calculations should exclude (unconditioned) attic roof overhangs, such as in the _SmallOffice_ prototype model. The calculated skylight area `target` is adjusted if attic overhangs are detected. No overhangs (i.e. `graX/gra0` equals unity)? No adjustment.